cargo/cargo_build_script.bzl

```diff
@@ -5,16 +5,33 @@ load("@io_bazel_rules_rust//rust:private/utils.bzl", "find_toolchain")
 load("@io_bazel_rules_rust//rust:rust.bzl", "rust_binary")
 
 def _expand_location(ctx, env, data):
-    for directive in "$(execpath ", "$(location ":
+    """[summary]
+
+    Args:
+        ctx (ctx): [description]
+        env ([type]): [description]
+        data ([type]): [description]
+
+    Returns:
+        [type]: [description]
+    """
+    for directive in ("$(execpath ", "$(location "):
         if directive in env:
             # build script runner will expand pwd to execroot for us
             env = env.replace(directive, "${pwd}/" + directive)
     return ctx.expand_location(env, data)
 
-def _expand_locations(ctx):
-    "Expand $(execpath ...) references in user-provided env vars."
-    env = ctx.attr.build_script_env
-    data = getattr(ctx.attr, "data", [])
+def _expand_locations(ctx, env, data):
```

Comment:
As above, except env is a dict whose values we iterate over. I'd probably only bother documenting this function - the other is a trivial helper with identical args.